### PR TITLE
add codecov config

### PR DESCRIPTION
**Pull request by LiZhenCheng9527**
**What type of PR is this?**

**What this PR does / why we need it**:
add codecov config
**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by LiZhenCheng9527** [comment]
/cc @hzxuzhonghu Please update CODECOV_TOKEN refer to https://docs.codecov.com/docs/quick-start#step-4-upload-coverage-reports-to-codecov

This is Kmesh codecov TOKEN：https://app.codecov.io/gh/kmesh-net/kmesh/settings

File: .codecov.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+coverage:
+  status:
+    project: #add everything under here, more options at https://docs.codecov.com/docs/commit-status
+      default:
+        # basic
+        target: auto #default
+        threshold: 35%
```
Comment (hzxuzhonghu) [inline .codecov.yml:7]: does 35% apply to new added code?

Comment (LiZhenCheng9527) [reply .codecov.yml:7]: It's a threshold.
Now the overall UT coverage of the Kmesh is36.06%.
If a new commit causes project coverage to fall below 35%, an alarm is raised.

File: .codecov.yml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+coverage:
+  status:
```
Comment (hzxuzhonghu) [inline .codecov.yml:2]: we need comment https://docs.codecov.com/docs/pull-request-comments

**Pull request merged**